src/core.ts

```diff
@@ -153,7 +156,7 @@ declare module "next/router" {
     [K in Route as K["pathname"]]: Exclude<K["query"], undefined>;
   };
 
-  type RouteOrQuery = Route | { query: { [key: string]: string | undefined } };
+  type RouteOrQuery = Route | { query?: UnknownQueryParams };
```

Comment:
Not sure if `query` was intentionally not optional here, but I changed it to align with the `next/link` one.

Reply:
It was, my thinking was that if you don't provide Route, you should provide a query, thus making `{}` an invalid argument. I assumed `{}` was rare enough that it was most likely an accident, and users could cast if it was intentional.

I don't feel particularly strongly about this stance thought